tests/test_laziness.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2022 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+import shutil
+
+from streaming import MDSWriter, StreamingDataset
+
+
+def test_laziness():
+    num_samples = 100_000
+    local = 'my-local'
+    remote = 'my-remote'
+    columns = {'value': 'int'}
+    compression = None
+    hashes = None
+    size_limit = 10_000
+
+    with MDSWriter(remote, columns, compression, hashes, size_limit) as out:
+        for i in range(num_samples):
+            sample = {'value': i}
+            out.write(sample)
+
+    # Verify __getitem__() downloads/accesses.
```

Comment:
This one doesn't verify any downloading right? because you've passed `remote` as the first arg, so it only uses the already existing files there?

Reply:
updated thanks